src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionResolver.cs

```diff
@@ -103,36 +125,119 @@ public GenerateMethodCodeActionResolver(DocumentContextFactory documentContextFa
+        {
+            // There is no existing @code block. This means that there is no code block source mapping in the generated C# document
+            // to place the code, so we cannot utilize the document mapping service and the formatting service.
+            // We are going to arbitrarily place the method at the end of the class in the generated C# file to
+            // just get the simplified text that comes back from Roslyn.
+
+            var classLocationLineSpan = @class.GetLocation().GetLineSpan();
+            var insertPosition = new Position(classLocationLineSpan.EndLinePosition.Line, 0);
+            var tempTextEdit = new TextEdit()
+            {
+                NewText = editToSendToRoslyn.NewText,
+                Range = new Range() { Start = insertPosition, End = insertPosition }
+            };
+
+            var delegatedParams = new DelegatedSimplifyMethodParams(documentContext.Identifier, RequiresVirtualDocument: true, tempTextEdit);
+            var result = await _languageServer.SendRequestAsync<DelegatedSimplifyMethodParams, TextEdit[]?>(
+                RazorLanguageServerCustomMessageTargets.RazorMethodEndpointName,
+                delegatedParams,
+                cancellationToken).ConfigureAwait(false);
+
+            if (result is not null)
+            {
+                // Roslyn will have removed the beginning formatting, put it back.
+                var formatting = editToSendToRoslyn.NewText[0..editToSendToRoslyn.NewText.IndexOf("private")];
+                editToSendToRoslyn.NewText = $"{formatting}{result.First().NewText.TrimEnd()}";
+            }
+        }
+        else if (_documentMappingService.TryMapToGeneratedDocumentRange(code.GetCSharpDocument(), editToSendToRoslyn.Range, out var remappedRange))
+        {
+            // If the call to Roslyn is successful, the razor formatting service will format incorrectly if our manual formatting is present,
+            // strip our manual formatting from the method so we just have a valid method signature.
+            var unformattedMethodSignature = templateWithMethodSignature
+                .Replace(FormattingUtilities.InitialIndent, string.Empty)
+                .Replace(FormattingUtilities.Indent, string.Empty);
+
+            var remappedEdit = new TextEdit()
+            {
+                NewText = unformattedMethodSignature,
+                Range = remappedRange
+            };
+
+            var delegatedParams = new DelegatedSimplifyMethodParams(documentContext.Identifier, RequiresVirtualDocument: true, remappedEdit);
+            var result = await _languageServer.SendRequestAsync<DelegatedSimplifyMethodParams, TextEdit[]?>(
+                RazorLanguageServerCustomMessageTargets.RazorMethodEndpointName,
+                delegatedParams,
+                cancellationToken).ConfigureAwait(false);
+
+            if (result is not null)
```

Comment:
Unfortunately because of the possibility of `result = null`, I did not remove a lot of the formatting code in `CodeBlockService`. If `result = null`, we should still produce a nicely formatted event handler, but we cannot use the `RazorFormattingService` at that point since we do not have C# edits, so we still do manual formatting work.